contracts/dao-core/src/contract.rs

```diff
@@ -845,9 +845,18 @@ pub fn query_proposal_module_count(deps: Deps) -> StdResult<Binary> {
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    let ContractVersion { version, .. } = get_contract_version(deps.storage)?;
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
     match msg {
         MigrateMsg::FromV1 { dao_uri } => {
+            // `CONTRACT_VERSION` here is from the data section of the
+            // blob we are migrating to. `version` is from storage. If
+            // the version in storage matches the version in the blob
+            // we are not upgrading.
+            if version == CONTRACT_VERSION {
```

Comment:
nit: thoughts on putting this check outside of the match statement and adding for all migrate methods?

Reply:
great question. i think "philosophically" it's not clear to me that all actions executed via the CosmWasm level admin stuff need to be about upgrading the code ID. in my ics721 contract i used this to make the ownership structure for config updates [simpler](https://github.com/public-awesome/ics721/blob/52852e07446c91983a91495e85103370f49814a7/contracts/cw-ics721-bridge/src/msg.rs#L180-L187), because CosmWasm has its own processes for updating an admin that i got for free.